php/pantheon/utils.php

```diff
@@ -60,8 +59,25 @@ public static function sanitize_data($data, $sanitizer_function = 'htmlspecialch
 				array_map('self::sanitize_data', array_values((array)$data))
 			);
 			return is_object( $data ) ? (object)$sanitized_data : $sanitized_data;
-		} elseif ( is_integer($data) ) {
+		} elseif ( is_integer( $data ) ) {
 			return (string)$data;
+		} elseif ( is_string( $data ) ) {
+			$dom = new \DOMDocument;
+			$dom->loadHTML( $data );
+			$anchors = $dom->getElementsByTagName('a');
+
+			// Bail if our string does not only contain an anchor tag.
+			if ( 0 === $anchors->length ) {;
+				return $sanitizer_function($data);
+			}
+
+			$href = $anchors[0]->getAttribute('href');
+			$sanitized_href = call_user_func($sanitizer_function, $href);
+			$sanitized_link_text = call_user_func($sanitizer_function, $anchors[0]->textContent);
+			
+			// Rebuild anchor tags to ensure there are no injected attributes.
+			$rebuilt_link = '<a href="' . $sanitized_href . '">' . $sanitized_link_text . '</a>';
```

Comment:
Need to confirm that this markup is accurate.

Reply:
confirmed here: https://github.com/pantheon-systems/wp_launch_check/blob/b40dd826c2fac66f91a453355f4612ba847fcca0/php/pantheon/checks/plugins.php#L57

updated markup in [99f4652](https://github.com/pantheon-systems/wp_launch_check/pull/141/commits/99f4652be8b2dcfe91f8f211746e5330a9cf5661)